### DISCUSSION TITLE
add entries to NXcollection with dict

### DISCUSSION THREAD

**Discussion by lrlunin** [nexpy/nexpy#392] [Q&A]
Hello,

in the docs I found that I am free to add entries (for example `NXfield`s) to the `NXcollection` via `entries` arguments which gets dict. Here is the docs link: https://nexpy.github.io/nexpy/treeapi.html#nexusformat.nexus.tree.NXgroup

I tried to do so
`q = NXentry(name="entry394", measurement=NXcollection(pumped=NXcollection(entries={'amplitude': np.linspace(0, 10, 10), 'center' : np.linspace(2, 12, 10)})))`
and then I get an error
`NeXusError: Keyword arguments must be valid NXobjects` .

Okay, it's not clear what my objects are. Then I need to wrap these `numpy` arrays in `NXfield`s. If I do so it works but the dictionary's keys will be ignored.
```python
q = NXentry(name="entry394", measurement=NXcollection(pumped=NXcollection(entries={'amplitude': NXfield(np.linspace(0, 10, 10)), 'center' : NXfield(np.linspace(2, 12, 10))})))

print(q.tree)
entry394:NXentry
  measurement:NXcollection
    pumped:NXcollection
      unknown = float64(10)
      unknown = float64(10)
```

 In this case I need to specify the keys in `name` arguments of `NXfield`s and it works fine. Does it work as expected and can I somehow pass the dict with keys?

**Comment by lrlunin** [comment]
And why it is not possible just to give a `list` of `NXfield`s in `entries` instead if the keys anyway will be overwritten by `name` of `NXfield`?
It would be much easier then to do so:
`q = NXentry(name="entry394", measurement=NXcollection(pumped=NXcollection(entries=[NXfield(np.linspace(0, 10, 10), name="amplitude"), NXfield(np.linspace(2, 12, 10), name="center")])))`

**Comment by rayosborn** [comment]
Thank you for raising this issue. At first glance, I think it would indeed make sense to convert values in the entries dictionary automatically to NXfields. I don't normally use this form of initializing NXgroups because the following works just as well:
```
>>> q = NXentry(name="entry394", measurement=NXcollection(pumped=NXcollection(amplitude=np.linspace(0, 10, 10), center=np.linspace(2, 12, 10))))
>>> print(q.tree)
entry394:NXentry
  measurement:NXcollection
    pumped:NXcollection
      amplitude = float64(10)
      center = float64(10)
```
However, that doesn't mean that your example shouldn't also be supported, especially as the documentation implies that it should work.

**Comment by rayosborn** [comment]
This should be an easy fix. I will convert this discussion to an issue so that it can be associated with the relevant PR.

**Comment by rayosborn** [comment]
This is really an issue with the [nexusformat](https://github.com/nexpy/nexusformat) package, rather than NeXpy. It is being fixed by [this PR](https://github.com/nexpy/nexusformat/pull/193).

**Comment by rayosborn** [comment]
This issue should have been fixed with the release of v1.0.1 of the nexusformat package. Note that you have to update `nexusformat`, not `nexpy`, to get the new version. 
```
$ pip install -U nexusformat
```
or
```
$ conda update nexusformat
```

**Reply by lrlunin** [reply]
Thank you very much for your help. I'll test it and report.

**Comment by lrlunin** [comment]
The test was rather delayed but now it works! Thank you ;-)
